run_unittests.sh

```diff
@@ -32,7 +34,8 @@ fi
 # FIXME: find a way to have nosetests recurse, or move the tests into
 #        a directory structure that nose can handle.
 for unittest in $(find subtests -name 'test_*.py'); do
-    nosetests -v $unittest
+    echo $unittest
+    nosetests -v $unittest || exit 1
```

Comment:
Thinking out loud:  I've since discovered some important things for us to consider.  (1) nosetests is unmaintained / forked to nosetests2.  (2) We're already depending on unittest2 which has it's own test discovery/runner script that does basically the same thing.  Consider replacing ``nosetests ...`` with ``unit2 ...`` (I think most of the options are even the same).  This would let us drop a dependency that will be disappearing in the future anyway.